rest/model/status.go

```diff
@@ -47,6 +47,71 @@ func (apiStatus *APIRecentTaskStats) ToService() (interface{}, error) {
 	return nil, errors.Errorf("ToService() is not implemented for APIRecentTaskStats")
 }
 
+type APIResult struct {
+	Name  APIString `json:"name"`
+	Count int       `json:"count"`
+}
+
+type APIRecentTaskStatsList struct {
+	Total              []APIResult `json:"total"`
+	Inactive           []APIResult `json:"inactive"`
+	Unstarted          []APIResult `json:"unstarted"`
+	Started            []APIResult `json:"started"`
+	Succeeded          []APIResult `json:"succeeded"`
+	Failed             []APIResult `json:"failed"`
+	SetupFailed        []APIResult `json:"setup-failed"`
+	SystemFailed       []APIResult `json:"system-failed"`
+	SystemUnresponsive []APIResult `json:"system-unresponsive"`
+	SystemTimedOut     []APIResult `json:"system-timed-out"`
+	TestTimedOut       []APIResult `json:"test-timed-out"`
+}
+
+func (s *APIRecentTaskStatsList) BuildFromService(h interface{}) error {
+	switch v := h.(type) {
+	case *task.ResultCountList:
+		for _, result := range v.Total {
+			s.Total = append(s.Total, APIResult{Name: ToAPIString(result.Name), Count: result.Count})
+		}
```

Comment:
I'd write a method that converts a result to a different kind of result, rather than having a bunch of loops here... 

as I read this code review the more I'm struck with the cost to adding a new status that this creates in evergreen. Might there be a different way of doing this? 

Perhaps we should return a thing that is just an array of { Status, Name, Count} documents rather than a map[Status]<name,count>

Reply:
reflection!

Reply:
> I'd write a method that converts a result to a different kind of result, rather than having a bunch of loops here...

done.

An array of {Status, Name, Count} would seem to me harder for the user to parse. I commented with a sample output. But yes, this will make more changes should we add a status and want stats about it.